src/Paramore.Brighter/CommandProcessor.cs

```diff
@@ -461,6 +472,166 @@ public async Task PublishAsync<T>(
             }
         }
 
+        /// <inheritdoc />
+        public string SchedulerPost<TRequest>(TRequest request, 
+            TimeSpan delay, 
+            RequestContext? requestContext = null,
+            Dictionary<string, object>? args = null) where TRequest : class, IRequest
+        {
+            if (_messageSchedulerFactory == null)
+            {
+                throw new InvalidOperationException("No message scheduler factory defined.");
+            }
+            
+            s_logger.LogInformation("Scheduling a request: {RequestType} {Id}", request.GetType(), request.Id);
+            
+            var span = _tracer?.CreateSpan(CommandProcessorSpanOperation.Scheduler, request, requestContext?.Span, options: _instrumentationOptions);
+            var context = InitRequestContext(span, requestContext);
+
+            try
+            {
+                Message message = s_mediator!.CreateMessageFromRequest(request, context);
+
+                var scheduler = _messageSchedulerFactory.Create(this);
```

Comment:
Let's sync up scheduler types to the path type; if you don't have the correct one, we assert. If you need to wrap async over sync or vice-versa, do it in the wrapper for the scheduler we are abstracting